### ISSUE TITLE
Extend KCC Bigtable support to include Table, Application Profiles and GC Policy

### ISSUE THREAD

**Issue by goldfishy** [GoogleCloudPlatform/k8s-config-connector#236]
Would it be possible to extend the KCC support for Bigtable to also include primitives managing:

- [Tables, Column Families](https://cloud.google.com/bigtable/docs/cbt-reference#create_a_table)
- [Application Profiles](https://cloud.google.com/bigtable/docs/cbt-reference#create_app_profile_for_an_instance)
- [GC Policies](https://cloud.google.com/bigtable/docs/cbt-reference#set_the_garbage-collection_policy_age_versions_for_a_column_family)

**Comment by maqiuyujoyce** [comment]
Thank you for your suggestion, @goldfishy ! We've added the three resources to the list of resources we're looking into and will let you know when we have more information.

**Comment by goldfishy** [comment]
Thanks for the quick reply @maqiuyujoyce !

Looking forward to hearing a rough timeline. :)

**Comment by maqiuyujoyce** [comment]
Hi @goldfishy , could you share the importance and urgency of these resources? Are they blockers for your project?

**Comment by maqiuyujoyce** [comment]
Hi @anmoel , after some internal discussion, we determined that supporting recovery of deleted resources is out of the scope because the purpose of Config Connector is to make GCP resource management a Kubernetes native experience, and in Kubernetes, deletions are really intended to be one-way. We recommend using `gcloud` if you need to recover your projects.

Additionally, "undeleting a project" is considered not part of the normal workflow, but only for emergent rollbacks. Could you provide more context about your scenario? Is it a reason you want to be able to recover the projects via Config Connector?

**Comment by xvzf** [comment]
Hello,

Blocker for our projects :-/ any ETA?

**Comment by kibbles-n-bytes** [comment]
Hey @xvzf , we have a team member currently working on including these. ETA for support for all three is roughly end of August.

**Comment by xvzf** [comment]
If you need an early adaptor, ping me 👍 Thx!

**Comment by spew** [comment]
BigtableTable is available in 1.17.0.

**Comment by xvzf** [comment]
Thanks for the update @spew !

**Comment by caieo** [comment]
BigtableAppProfile is available in 1.18.0. BigtableGCPolicy should be coming in by the next release.

**Comment by goldfishy** [comment]
Thank you @caieo @spew ! 

Super helpful.

**Comment by jcanseco** [comment]
Hi all, `BigtableGCPolicy` is now available in 1.19.0.

**Comment by xvzf** [comment]
@spew The IAMPolicies do not allow to reference a Table. Can this be addressed?

**Comment by maqiuyujoyce** [comment]
Hi @xvzf , thank you for your request. We are looking into prioritizing it and will let you know if there is any update.

**Comment by xvzf** [comment]
Hi @maqiuyujoyce, is there any ETA yet? It is a blocker for automation right now

**Comment by xiaobaitusi** [comment]
Hi @xvzf, sorry for your inconvenience. We have added it into our near-term backlog; we'll start working on it once our dependency lib is updated with iam support for tables. Thanks for your patience.

**Comment by xvzf** [comment]
Hey @xiaobaitusi is there any update?

**Comment by H4nks** [comment]
Hey @xiaobaitusi, any update yet ? Support of IAM Policies is a blocker for us too I'm afraid :(

**Comment by jcanseco** [comment]
Hi @xvzf @H4nks, thanks, we just got unblocked by a dependency, and so we should be able to get to this item by end of Nov / early Dec.

**Comment by H4nks** [comment]
That's awesome news ! Thank you very much @jcanseco and @caieo !

**Comment by jcanseco** [comment]
No worries! Credit to @zeya for implementation :)

**Comment by AvivMataraso** [comment]
there is option to also support the creation of multiple column families with GC policy? or create a bigtable table with families with GC policy? 
(there is only option to create multiple column families in the creation of the the table, but without GC policy).
